native/src/contexts/AppContextProvider.tsx

```diff
@@ -28,15 +30,26 @@ const AppContextProvider = ({ children }: AppContextProviderProps): ReactElement
   const [languageCode, setLanguageCode] = useState<string | null>(null)
 
   const loadSettings = useCallback(async () => {
-    const { selectedCity, contentLanguage } = await appSettings.loadSettings()
+    const settings = await appSettings.loadSettings()
+    const { selectedCity, contentLanguage } = settings
     if (!contentLanguage) {
       throw new Error('Language not initialized by I18nProvider!')
     }
     setCityCode(selectedCity)
     setLanguageCode(contentLanguage)
+    return settings
   }, [])
 
   const settingsResponse = useLoadAsync(loadSettings)
+  const allowPushNotifications = settingsResponse.data?.allowPushNotifications ?? null
+
+  useEffect(() => {
+    if (cityCode) {
+      dataContainer.storeLastUsage(cityCode).catch(reportError)
+    }
+  }, [cityCode])
+
+  useSubscribePushNotifications({ cityCode, languageCode, allowPushNotifications })
```

Comment:
Not sure if this is the right place for this, let me know if you think it should be moved somewhere else.